src/components/narrative/index.js

```diff
@@ -52,15 +52,16 @@ class Narrative extends React.Component {
         }));
         return;
       }
-
-      this.props.dispatch(changePage({
-        // path: this.props.blocks[blockIdx].dataset, // not yet implemented properly
-        changeDataset: false,
+      const change = {
         query: queryString.parse(this.props.blocks[idx].query),
         queryToDisplay: {n: idx},
         push: true
-      }));
-
+      };
+      if (this.props.blocks[idx].dataset !== this.props.blocks[this.props.currentInFocusBlockIdx].dataset) {
```

Comment:
Is this the only section which needs to be transferred to the mobile narrative display for that to work?